### PR TITLE
Tdl 18167 check account access

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
- Added a check account call after checking the access token
- Raises an error if the account check fails

# Manual QA steps
 - Verify that the tap throws an error if invalid accounts are provided
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_tiktok_ads/client.py [hunk 1]
```diff
@@ -58,15 +60,24 @@ def check_access_token(self):
...
-            return False
+            raise Exception('Error status_code = %s', response.status_code)
         resp = response.json()
         error_code = resp.get('code')
         message = resp.get('message', 'Unknown Error occurred.')
 
         if error_code != 0: # `0` error code indicates successful request
             raise TikTokAdsClientError(message, response) # raise the exception with the message retrieved
-        return bool(resp.get('message') == 'OK')
+        if resp['message'] == 'OK':
```
Comment (dbshah1212) [inline tap_tiktok_ads/client.py:70]: Add code comments here

File: tap_tiktok_ads/client.py [hunk 2]
```diff
@@ -58,15 +60,24 @@ def check_access_token(self):
...
+            self.__verified = True
+            headers = {
+                "Access-Token": self.__access_token
+            }
+            params = {
+                "advertiser_ids": self.__advertiser_id
+            }
+            adv_response = self.get(path='advertiser/info/', headers=headers,
```
Comment (dbshah1212) [inline tap_tiktok_ads/client.py:78]: Add code comments to explain this call

File: tap_tiktok_ads/client.py (hunk 1 above)
Comment (namrata270998) [reply tap_tiktok_ads/client.py:73]: Added comments

File: tap_tiktok_ads/client.py [hunk 3]
```diff
@@ -67,7 +69,19 @@ def check_access_token(self):
...
+        # if the check_access_token() succeeds, then check the account access with the account ids provided in config.
+        if resp['message'] == 'OK':
+            self.__verified = True
+            headers = {
+                "Access-Token": self.__access_token
+            }
+            params = {
+                "advertiser_ids": self.__advertiser_id
+            }
+            # Call the advertisers API with the account ids to check whether the accounts ar valid or not.
```
Comment (dbshah1212) [inline tap_tiktok_ads/client.py:81]:
```suggestion
            # Call the advertisers API with the account ids to check whether the accounts are valid or not.
```

File: tap_tiktok_ads/client.py [hunk 4]
```diff
@@ -67,7 +69,19 @@ def check_access_token(self):
...
+            adv_response = self.get(path='advertiser/info/', headers=headers,
```
Comment (dbshah1212) [inline tap_tiktok_ads/client.py:82]: We need to do separate calls for each account_id else How do we get to know the culprit?

Comment (namrata270998) [reply tap_tiktok_ads/client.py:82]: The API itself returns which advertiser_ids are invalid
So we won't need individual calls

File: tap_tiktok_ads/client.py (hunk 3 above)
Comment (namrata270998) [reply tap_tiktok_ads/client.py]: fixed the typo

**Review by dbshah1212 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**